core/templates/pages/exploration-player-page/services/content-translation-manager.service.ts

```
@@ -121,22 +128,27 @@ export class ContentTranslationManagerService {
       languageCode: string,
       content: SubtitledHtml|SubtitledUnicode
   ): void {
-    const writtenTranslation = writtenTranslations.translationsMapping[
-      content.contentId][languageCode];
+    const writtenTranslation: WrittenTranslation | null = (
+      content.contentId ? writtenTranslations.translationsMapping[
+        content.contentId][languageCode] : null);
 
-    if (!this._isValidStringTranslation(writtenTranslation)) {
+    if (!writtenTranslation ||
+      !this._isValidStringTranslation(writtenTranslation)
+    ) {
       return;
     }
 
-    let valueName;
+    let valueName = '';
     // Note: The content can only be of type SubtitledHtml|SubtitledUnicode.
     if (content instanceof SubtitledHtml) {
-      valueName = 'html';
+      valueName = '_html';
     } else if (content instanceof SubtitledUnicode) {
-      valueName = 'unicode';
+      valueName = '_unicode';
```

Should we throw an error if the content is of a different type?

---

I don't believe that's necessary since the `content` parameter type is `SubtitledHtml | SubtitledUnicode` so the argument can only be either of the two.
https://github.com/oppia/oppia/blob/a4a3c2b7c673e928264919b595f51222ddd8e2c5/core/templates/pages/exploration-player-page/services/content-translation-manager.service.ts#L119-L123